src/MacVim/gui_macvim.m

```diff
@@ -285,14 +310,7 @@
         [[MMBackend sharedInstance] addToMRU:filenames];
     }
 
-#if MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_10
-    {
-	NSOperatingSystemVersion version = {10, 13, 0};
-
-	is_macos_high_sierra_or_later =
-	    [[NSProcessInfo processInfo] isOperatingSystemAtLeastVersion:version];
-    }
-#endif
+    check_system_version();
```

Comment:
if ([NSProcessInfo instancesRespondToSelector:@selector(isOperatingSystemAtLeastVersion:)]) {
      NSOperatingSystemVersion version = {10, 13, 0};
      is_macos_high_sierra_or_later = [[NSProcessInfo processInfo] isOperatingSystemAtLeastVersion:version];	
    }

Reply:
Thank you, I updated.

Reply:
Sorry, I meant we don't need check_system_version method, specially the SystemVersion.plist part since isOperatingSystemAtLeastVersion: is available on 10.10 or later.

Reply:
When build MacVim with env `MACOSX_DEPLOYMENT_TARGET` < 10.10 (.travis.yml has set `=10.8`) `isOperatingSystemAtLeastVersion:` isn't available, so cannot build.
Therefore SystemVersion.plist part is needed in order to use MacVim built with env `MACOSX_DEPLOYMENT_TARGET` < 10.10 on 10.13.

Reply:
I don't think we need to support that situation. Using SDK 10.9 or earlier and run the binary on 10.13 or later. It's overkill. We do support the latest Xcode + the latest SDK, and the binary could run on 10.8. Probably people can use SDK 10.8 for building MacVim for 10.8, but not for 10.13.

Reply:
Actually .travis.yml has set `MACOSX_DEPLOYMENT_TARGET=10.8` so release-version (snapshot) MacVim is built on SDK 10.8.
Thus snapshot 146 does not include the processing of checking-10.13 (below) and reproduce #620 .
https://github.com/macvim-dev/macvim/blob/54c198e2d18660e8d880b67cbf01f3aae52896fe/src/MacVim/gui_macvim.m#L288-L295

Reply:
So the minimal code would look like

    }
    
    #if MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_10
        if ([[NSProcessInfo processInfo]
                respondsToSelector:@selector(isOperatingSystemAtLeastVersion:)])
        {
            NSOperatingSystemVersion version = {10, 13, 0};

            is_macos_high_sierra_or_later =
                [[NSProcessInfo processInfo] isOperatingSystemAtLeastVersion:version];
        }
    #endif

        return OK;

Reply:
On travis build (`MACOSX_DEPLOYMENT_TARGET=10.8`), `MAC_OS_X_VERSION_MAX_ALLOWED` is `MAC_OS_X_VERSION_10_8` so that code block is dropped.

Reply:
Current master .travis.yml uses xcode9.2, so `MAC_OS_X_VERSION_MAX_ALLOWED` would be 10.12 or 10.13, and `MAC_OS_X_VERSION_MIN_REQUIRED` is 10.8 as `MACOSX_DEPLOYMENT_TARGET` in .travis.yml.

Reply:
Oh, `MAC_OS_X_VERSION_MAX_ALLOWED`... I saw wrong. OK, I understand.

Reply:
I updated.